pkg/provision/primitives/public_ip.go

```diff
@@ -66,7 +66,7 @@ func (p *Provisioner) publicIPProvisionImpl(ctx context.Context, reservation *pr
 
 	tapName := fmt.Sprintf("p-%s", reservation.ID) // TODO: clean this up, needs to come form networkd
 	fName := filterName(reservation.ID)
-	mac := ifaceutil.HardwareAddrFromInputBytes(config.IP.IP.To4())
+	mac := ifaceutil.HardwareAddrFromInputBytes([]byte(reservation.ID))
```

Comment:
whats the reason for this change?

Reply:
The reason for this is that the mac need to be set also on the macvtap device on host. networkd only take the reservation ID to create the macvtap device. so instead of changing the zbus interface instead I changed the mac to use the reservation ID instead. this way 
- the macvtap has the correct mac
- the CH is passed the correct mac
- the vm gets (of course) the same mac
if the mac on the host does not match the one in the vm it will not work